specification/iotcentral/data-plane/Microsoft.IoTCentral/preview/1.2-preview/iotcentral.json

```diff
@@ -0,0 +1,5638 @@
+          },
+          {
+            "$ref": "#/parameters/ComponentName"
+          },
+          {
+            "$ref": "#/parameters/CommandName"
+          },
+          {
+            "in": "body",
+            "name": "body",
+            "description": "Device command body.",
+            "schema": {
+              "$ref": "#/definitions/DeviceCommand"
+            },
+            "required": true
+          }
+        ],
+        "responses": {
+          "201": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/DeviceCommand"
+            }
+          },
+          "default": {
+            "description": "Error",
+            "schema": {
+              "$ref": "#/definitions/Error"
+            }
+          }
+        }
+      }
+    },
+    "/devices/{deviceId}/components/{componentName}/properties": {
+      "get": {
+        "operationId": "Devices_GetComponentProperties",
+        "summary": "Get device properties for a specific component",
+        "x-ms-examples": {
+          "Get properties on components": {
+            "$ref": "./examples/devices_properties_components_get.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/Version"
+          },
+          {
+            "$ref": "#/parameters/DeviceId"
+          },
+          {
+            "$ref": "#/parameters/ComponentName"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/DeviceProperties"
+            }
+          },
+          "default": {
+            "description": "Error",
+            "schema": {
+              "$ref": "#/definitions/Error"
+            }
+          }
+        }
+      },
+      "put": {
+        "operationId": "Devices_ReplaceComponentProperties",
+        "summary": "Replace device properties for a specific component",
+        "x-ms-examples": {
+          "Update properties on components": {
+            "$ref": "./examples/devices_properties_components_set.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/Version"
+          },
+          {
+            "$ref": "#/parameters/DeviceId"
+          },
+          {
+            "$ref": "#/parameters/ComponentName"
+          },
+          {
+            "in": "body",
+            "name": "body",
+            "description": "Device properties.",
+            "schema": {
+              "$ref": "#/definitions/DeviceProperties"
+            },
+            "required": true
+          }
+        ],
+        "responses": {
+          "202": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/DeviceProperties"
+            }
+          },
+          "default": {
+            "description": "Error",
+            "schema": {
+              "$ref": "#/definitions/Error"
```

Comment:
Operations with a 202 response should specify `x-ms-long-running-operation: true`.